sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -2176,4 +2176,32 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-39355: Avoid UnresolvedAttribute.apply throwing ParseException") {
+    checkAnswer(
+      sql("""
+            |SELECT *
```

Comment:
I'm surprised this bug causes real query failures. Can you explain how this query triggers `UnresolvedAttribute.apply`?

Reply:
A `Cast` may be generated `InConversion`, at which time `newSub.output` will call `UnresolvedAttribute.apply`.
https://github.com/apache/spark/blob/d3d22928d4fca1cd71f96a1308f0cc5d00120ad5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala#L349-L360

Reply:
I think we should fix `Alias.toAttribute`. It should create an attribute to reference the output of this alias, which is a single column, and we need to use `UnresolvedAttribute.quoted`.

Reply:
It seems that there may be a problem here, although I didn't reproduce it using SQL, do I need to fix it?

https://github.com/apache/spark/blob/2349175e1b81b0a61e1ed90c2d051c01cf78de9b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala#L120

Reply:
yea we should also use `.quoted` there. good catch!